script.js

```diff
@@ -0,0 +1,338 @@
+}
+// THE CALCULATOR CODE STARTS HERE
+let clearButton = document.querySelector(".allClear")
+let deleteButton = document.querySelector(".deleteBtn")
+// let calculatorDisplay = document.getElementById("calculatorDisplay");
+let secondDisplayContainer = document.querySelector(".secondDisplayContainer");
+let signs = document.querySelector(".signs");
+// let decimalPoint = document.querySelector(".decimalPoint")
+let NumberButton = document.querySelectorAll(".operands")
+let firstDisplay = document.querySelector(".first-display")
+let dummyDisplay = document.querySelector(".dummy-display")
+let add = document.querySelector(".additionBtn")
+let subtract = document.querySelector(".subtractionBtn")
+let multiply = document.querySelector(".multiplicationBtn")
+let divide = document.querySelector(".divisionBtn")
+let operationButton = document.querySelector(".equalsToBtn")
+let secondDisplay = document.createElement("div");
+NumberButton.forEach(button => {
+   button.addEventListener("click",eachnumber)
+    function eachnumber(){
+      firstDisplay.textContent += button.textContent
+      add.disabled = false
+      subtract.disabled = false
+      multiply.disabled = false
+      divide.disabled = false
+    }
+});
+add.addEventListener("click",added)
+subtract.addEventListener("click",subtracted)
+multiply.addEventListener("click",multiplied)
+divide.addEventListener("click",divided)
+decimalPoint.addEventListener("click", decima)
+operationButton.addEventListener("click",operate)
+clearButton.addEventListener("click",allclear)
+deleteButton.addEventListener("click",deleteCurrentDisplay)
+function decima(){
+    firstDisplay.textContent += "."
+    if(firstDisplay.textContent.includes(".")){
+        decimalPoint.disabled = true
+    }
+    else{
+        decimalPoint.disabled = false
+    }
+}
+function deleteCurrentDisplay(){
+ firstDisplay.textContent = firstDisplay.textContent.substring(0, firstDisplay.textContent.length -1);
+}
+function allclear(){
+    firstDisplay.textContent = ""
+    secondDisplay.textContent = ""
+    signs.textContent = ""
+    decimalPoint.disabled = false
+    add.disabled = true
+    divide.disabled = true
+    subtract.disabled = true
+    multiply.disabled = true
+}
+add.disabled = true
+function added(){
+disableButtonRepeat()
+    if(firstDisplay.textContent !=="" && secondDisplay.textContent !== "" && signs.textContent !== ""){
+        calculate()
+        secondDisplay.textContent = firstDisplay.textContent
+        signs.textContent = "+"
+        firstDisplay.textContent = ""
+    }
+    else{
+        secondDisplayContainer.appendChild(secondDisplay);
+        secondDisplay.classList.toggle("second-display");
+        dummyDisplay.classList.remove("dummy-display");
+        secondDisplay.textContent =  firstDisplay.textContent;
+        signs.textContent = "+"
+      if(firstDisplay.textContent === "")return
+        else{
+            clearFirstDisplay()
+        } 
```

Comment:
Since the only thing `clearFirstDisplay` does is: `firstDisplay.textContent = ""`, why not always call it here?